tests/test.bal

```diff
@@ -15,29 +15,478 @@
+    dependsOn: [
+        "testRevokePublisher",
+        "testGetRevokedPublishers"
+    ],
+    enable: true
+}
+function testResumePublisher() {
+    var result = publisherClient->resumePublisher("myeventhub", "device-1");
+    if (result is error) {
+        test:assertFail(msg = result.message());
+    }
+    test:assertTrue(result is ());
+    if (result is ()) {
+        log:print("successful");
+    }
+}
+
+@test:Config {
+    groups: ["consumergroup"],
+    enable: true
+}
+function testCreateConsumerGroup() {
+    var result = managementClient->createConsumerGroup("myeventhub", "consumerGroup1");
+    if (result is error) {
+        test:assertFail(msg = result.message());
+    }
+    test:assertTrue(result is xml);
+    if (result is xml) {
+        log:print(result.toString());
+    }
+}
+
+@test:Config {
+    groups: ["consumergroup"],
+    dependsOn: ["testCreateConsumerGroup"],
+    enable: true
+}
+function testGetConsumerGroup() {
+    var result = managementClient->getConsumerGroup("myeventhub", "consumerGroup1");
+    if (result is error) {
+        test:assertFail(msg = result.message());
+    }
+    test:assertTrue(result is xml);
+    if (result is xml) {
+        log:print(result.toString());
+    }
+}
+
+@test:Config {
+    groups: ["consumergroup"],
+    dependsOn: ["testCreateConsumerGroup"],
+    enable: true
+}
+function testListConsumerGroups() {
+    var result = managementClient->listConsumerGroups("myeventhub");
+    if (result is error) {
+        test:assertFail(msg = result.message());
+    }
+    test:assertTrue(result is xml);
+    if (result is xml) {
+        log:print("successful");
+        log:print(result.toString());
+    }
+}
+
+@test:Config {
+    groups: ["consumergroup"],
+    dependsOn: ["testCreateConsumerGroup"],
+    enable: true
+}
+function testListPartitions() {
+    var result = managementClient->listPartitions("myeventhub", "consumerGroup1");
+    if (result is error) {
+        test:assertFail(msg = result.message());
+    }
+    test:assertTrue(result is xml);
+    if (result is xml) {
+        log:print("successful");
+        log:print(result.toString());
+    }
+}
+
+@test:Config {
+    groups: ["consumergroup"],
+    dependsOn: ["testCreateConsumerGroup"],
+    enable: true
+}
+function testGetPartition() {
+    var result = managementClient->getPartition("myeventhub", "consumerGroup1", 1);
+    if (result is error) {
+        test:assertFail(msg = result.message());
+    }
+    test:assertTrue(result is xml);
+    if (result is xml) {
+        log:print(result.toString());
+    }
+}
+
+@test:Config {
+    groups: ["consumergroup"],
+    dependsOn: [
+        "testCreateConsumerGroup",
+        "testGetConsumerGroup",
+        "testListConsumerGroups",
+        "testListPartitions",
+        "testGetPartition"
+    ],
+    enable: true
+}
+function testDeleteConsumerGroups() {
+    var result = managementClient->deleteConsumerGroup("myeventhub", "consumerGroup1");
```

Comment:
Should we use the event hub name as a constant so that we can share it across both setup and teardown stages? Does it make sense?

Reply:
Yes it makes sense. It would be easy to manage if we do so. I added the event hub names as constants so that we can share it across both setup and tear down stages. Fixed. Thank you.